src/elastio-snap.c

```diff
@@ -3987,12 +4030,25 @@ static int __tracer_setup_snap(struct snap_device *dev, unsigned int minor, stru
 	dev->sd_gd->flags |= GENHD_FL_NO_PART_SCAN;
 #endif
 
+#ifdef HAVE_GENHD_FL_NO_PART
+	// the flag has been renamed in 5.17
+	dev->sd_gd->flags |= GENHD_FL_NO_PART;
+#endif
```

Comment:
Why do we need a feature test for it and don't just
```suggestion
#ifdef GENHD_FL_NO_PART
	// the flag has been renamed in 5.17
	dev->sd_gd->flags |= GENHD_FL_NO_PART;
#endif
?
```